addons/purchase/models/purchase.py

```diff
@@ -125,6 +125,7 @@ def _compute_invoice(self):
     product_id = fields.Many2one('product.product', related='order_line.product_id', string='Product', readonly=False)
     user_id = fields.Many2one('res.users', string='Purchase Representative', index=True, tracking=True, default=lambda self: self.env.user)
     company_id = fields.Many2one('res.company', 'Company', required=True, index=True, states=READONLY_STATES, default=lambda self: self.env.company_id.id)
+    currency_rate = fields.Float("Currency Rate", compute='_compute_currency_rate', compute_sudo=True, store=True, digits=(12, 6), readonly=True, help='Ratio between the purchase order currency and the company currency')
```

Comment:
Do you really need the digits?

Reply:
default decimal precision is 3 digit that's why we have change it to 6 digits ( to more accurate ) same as like all places.

Reply:
https://github.com/odoo/odoo/blob/12.0/odoo/fields.py#L1237 default is float8 double precision, it's not needed to trunk it in this case